### PR TITLE
Multiple same category intervention

### PR DESCRIPTION
**Pull request by jiayuanz3**
[##](url) Summary

Extend the single intervention to allowing multiple same category of interventions. It's notable that there's no time overlap under each intervention type (i.e. the end time of first case isolation should be smaller than the start time of second case isolation). The case isolation intervention example is updated with two case isolation and it remains one intervention for the remaining interventions, but modify the codes to be suitable for current setting. 

## Further work

If relevant, add in what needs to be done next on this area (and whether you will do it!)

## Checklist

- [ ] All new functions have docstrings in the correct style
- [ ] I've verified the complete docs build locally without errors
- [ ] I've maintained 100% coverage (please mention any 'no cover' annotations explicitly)
- [ ] I've unit-tested all new methods directly

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have read the **CONTRIBUTING** document.
- [ ] My code follows the code style of this project.
- [ ] I have updated the wiki with new parameters/model functionality

## Closing issues

Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: pyEpiabm/pyEpiabm/sweep/intervention_sweep.py [hunk 1]
```diff
@@ -72,9 +79,16 @@ def __call__(self, time):
             num_cases = sum(map(lambda cell: cell.number_infectious(),
                             self._population.cells))
             if intervention.is_active(time, num_cases):
-                intervention(time)
                 if self.intervention_active_status[intervention] is False:
+                    # Update with the single intervention
+                    # (with respect to index)
+                    intervention_type = self.intervention_dict.inverse[
```
Comment (KCGallagher) [inline pyEpiabm/pyEpiabm/sweep/intervention_sweep.py:85]: I agree its unconventional but there are ways to get a key from a dict based on the value, especially not dicts are ordered https://stackoverflow.com/questions/8023306/get-key-by-value-in-dictionary

File: python_examples/intervention_example/intervention_outputs/case_isolation_Icurve_plot.png
[the remark this answers is not in the document]
Comment (KCGallagher) [reply python_examples/intervention_example/intervention_outputs/case_isolation_Icurve_plot.png:1]: (This applies to all graphs here)

**Review by KCGallagher — CHANGES_REQUESTED** [review]
Afraid I only had a chance to have a brief look - here are some initial thoughts but I will go though the mechanisms of how you handle the parameter lists more carefully

File: pyEpiabm/pyEpiabm/sweep/intervention_sweep.py (hunk 1 above)
Comment (jiayuanz3) [reply pyEpiabm/pyEpiabm/sweep/intervention_sweep.py]: replaced the bidict by the way you suggested

File: pyEpiabm/pyEpiabm/tests/test_func/helper_func.py [hunk 2]
```diff
@@ -87,3 +87,15 @@ def sweep_list_initialise(cls):
             pe.sweep.QueueSweep(),
             pe.sweep.HostProgressionSweep(),
         ]
+
+    @classmethod
+    def intervention_conversion_list(cls, intervention_type):
+        """If the intervention parameters is in the dictionary format,
+        then convert it to dictionary to make consistent with the
```
[the remark this answers is not in the document]
Comment (jiayuanz3) [reply pyEpiabm/pyEpiabm/tests/test_func/helper_func.py]: dictionary to list conversion*

File: python_examples/intervention_example/intervention_outputs/case_isolation_Icurve_plot.png
[the remark this answers is not in the document]
Comment (jiayuanz3) [reply python_examples/intervention_example/intervention_outputs/case_isolation_Icurve_plot.png:1]: They should be the same. I think the plots haven't been updated (i.e. change code or modify parameters) for a long while and I rerun all of them (as I changed intervention's format in parameter.json). Also, as Nathan modified R_0 in the current main, after merging, it still needs to rerun.

**Pull request closed without merging**